src/core/bittorrent/peerinfo.cpp

```diff
@@ -253,3 +253,25 @@ QString PeerInfo::connectionType() const
 
     return connection;
 }
+
+qreal PeerInfo::getRelevance(BitTorrent::TorrentHandle * const torrent) const
```

Comment:
This is incorrect!
Just calculate relevance when you create PeerInfo object (modify its constructor), and return the previously calculated value here.
PeerInfo is simple structure-like type with some data conversions (but without calculations etc.). Don't break it.
Besides, your code has incorrect logic. For example, we may receive PeerInfo from one of the torrent, and in relevance calculating set a completely different torrent.